src/services/suggestionDiagnostics.ts

```diff
@@ -104,4 +122,34 @@ namespace ts {
     function getErrorNodeFromCommonJsIndicator(commonJsModuleIndicator: Node): Node {
         return isBinaryExpression(commonJsModuleIndicator) ? commonJsModuleIndicator.left : commonJsModuleIndicator;
     }
+
+    function isPromiseType(T: Type): boolean {
+        return T.flags === TypeFlags.Object && T.symbol.name === "Promise";

```

Comment:
as i mentioned in a previous iteration, i do not think this is the correct check. we should instead expose `getPromisedTypeOfPromise` function, and check that the result is not `undefined`. this handles cases of PromiseLike, and aliases to the `Promise` type.